src/components/form/OneClickForm/components/DataField/inputs/DataFieldAddressInput.tsx

```diff
@@ -0,0 +1,164 @@
+import { useRef, memo, ReactElement, useMemo } from 'react';
+import { produce } from 'immer';
+import { Box, TextField, TextFieldProps } from '@mui/material';
+import { useFormContext } from 'react-hook-form';
+import isEqual from 'lodash/isEqual';
+
+import { fromUSAddress, toUSaddress } from '../../../utils/addressFormatter';
+import { inputStyle } from '../../../styles/input';
+
+import { extractChildrenFromCredentialFieldSet } from '../../CredentialsDisplay/utils';
+import { useCredentialsDisplayItem } from '../../CredentialsDisplay/CredentialsDisplayItemContext';
+import { useCredentialsDisplayItemValid } from '../../CredentialsDisplay/hooks';
+
+import { DataFieldLabelText } from '../DataFieldLabelText';
+import { DataFieldClearAdornment } from '../DataFieldClearAdornment';
+import { CredentialFieldSet } from '../../CredentialsDisplay/types';
+
+type DataFieldAddressInputMemoizedProps = {
+  credentialsDisplayItem: ReturnType<typeof useCredentialsDisplayItem>;
+  itemValid: ReturnType<typeof useCredentialsDisplayItemValid>;
+};
+
+const DataFieldAddressInputMemoized = memo(
```

Comment:
Please add an inline block comment explaining this function. It seems important and is pretty extensive!